Tasks/PublishBuildArtifacts/task.json

```diff
@@ -21,15 +21,15 @@
       "name": "PathtoPublish",
       "type": "filePath",
       "label": "Path to publish",
-      "defaultValue": "",
+      "defaultValue": "$(Build.ArtifactStagingDirectory)",
```

Comment:
Not all tasks will output to the artifact staging directory, so having a default that may not work is not a good idea. It is better to leave it empty since default of $(System.workingdirectory) is more likely to work for most tasks.

Reply:
I'll propose it's okay to default this value for when artifacts are in the **Build.ArtifactStagingDirectory** folder.  Then, people only have to change it if they depart from the prescribed pattern.  ArchiveFiles does this by defaulting its **rootFolderOrFile** input to `$(Build.BinariesDirectory)`.  When it comes to artifacts, the following tasks default to **Build.ArtifactStagingDirectory**: DotNetCoreCLI, DownloadBuildArtifacts, AppCenterTest, HelmDeploy, NuGetCommand, and DownloadPackage.  So this change would make PublishBuildArtifacts more in harmony with those too.  Also, our build templates already default the PublishBuildArtifacts task to use this value.

Reply:
If we want people to be selective with what they publish to the service, I think it would be good to either make $(Build.ArtifactStagingDirectory) the default path, or to add match patterns to this task. $(Build.ArtifactStagingDirectory) is better if the default match pattern would be `**\*`.

FWIW, if you publish $(Build.ArtifactStagingDirectory) without copying any files to it, your build succeeds with a warning from the agent: `##[warning]Directory 'D:\a\1\a' is empty. Nothing will be added to build artifact 'drop'.`